[WINDOWS] On startup from the zip distrubtion main.R is not run

## Description

For the zip distribution of the rattlng package for Windows, when running the rattle.exe on Windows, after R starts up in the CONSOLE it does not run the main.R script. On Linux and MacOS the main.R script is run on startup.

This task is to investigate why main.R is not run on starting up rattle.exe on Windows.

Unfortunately this fix causes the following error for Linux:

```
flutter run --device-id linux
Launching lib/main.dart on Linux in debug mode...
Building Linux application...                                   
/home/gjw/git/github/gjwgit/rattleng/linux/flutter/ephemeral/.plugin_symlinks/flutter_pty/src/flutter_pty_unix.c:114:5: error: call to undeclared function 'waitpid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
Error: Build process failed
```

Reverting for now.

--- a/lib/r/start.dart
+++ b/lib/r/start.dart
@@ -26,6 +26,7 @@
 
 library;
 
+import 'dart:async';
 import 'dart:convert';
 
 import 'package:flutter/material.dart';
@@ -99,7 +100,19 @@ void rStart(BuildContext context, WidgetRef ref) async {
   // startup. Seems to have a slight delay on Linux with a all black
   // screen. Let's see what it does on Windows.
 
-  await Future(() {
+  await Future(() async {
+    
+    // Create a Completer to ensure pty is ready before writing to it.
+    Completer<void> ptyReadyCompleter = Completer<void>();
+
+    // Mark the pty as ready after it has been initialized.
+    ref.read(ptyProvider).exitCode.then((_) {
+      ptyReadyCompleter.complete();
+    });
+
+    // extra delay
+    const duration = Duration(seconds: 1);
+    await Future.delayed(duration);
     // Add the code to the script.
 
     updateScript(ref, code);

